designer/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/simple/SimpleStateStatus.scala

```diff
@@ -15,4 +18,116 @@ object SimpleStateStatus {
   val Failed: StateStatus = FailedStateStatus("FAILED")
   val Error: StateStatus = AllowDeployStateStatus("ERROR")
   val Warning: StateStatus = AllowDeployStateStatus("WARNING")
+
+  val defaultActions: List[ProcessActionType] = Nil
+
+  val actionStatusMap: Map[ProcessActionType, StateStatus] = Map(
+    ProcessActionType.Deploy -> SimpleStateStatus.Running,
+    ProcessActionType.Cancel -> SimpleStateStatus.Canceled,
+    ProcessActionType.Archive -> SimpleStateStatus.NotDeployed,
+    ProcessActionType.UnArchive -> SimpleStateStatus.NotDeployed
+  )
+
+  val statusActionsMap: Map[StateStatus, List[ProcessActionType]] = Map(
+    SimpleStateStatus.Unknown -> List(ProcessActionType.Deploy),
+    SimpleStateStatus.NotDeployed -> List(ProcessActionType.Deploy, ProcessActionType.Archive),
+    SimpleStateStatus.DuringDeploy -> List(ProcessActionType.Deploy, ProcessActionType.Cancel), // Deploy? see FlinkStateStatus
+    SimpleStateStatus.Running -> List(ProcessActionType.Cancel, ProcessActionType.Pause, ProcessActionType.Deploy),
+    SimpleStateStatus.Canceled -> List(ProcessActionType.Deploy, ProcessActionType.Archive),
+    SimpleStateStatus.Restarting -> List(ProcessActionType.Deploy, ProcessActionType.Cancel), // Deploy? see FlinkStateStatus
```

Comment:
Is it ok to allow Deploy action in Restarting or DuringDeploy status? 
It is not allowed by FlinkStateStatus and that seems to be a correct behaviour.
@arkadius wdyt?

Reply:
Regarding "Restarting" it means that something not ok happens on engine side. Ideally would be the best to allow override this "not ok" state of engine, but AFAIK it is problematic on Flink so user have to cancel the previous job and after that deploy next job. Summing up it looks like Flink's quirk that deploy is not allowed in the restarting state